query-engine/prisma-models/src/composite_type.rs

```diff
@@ -32,23 +27,21 @@ impl CompositeType {
             .expect("Invalid back-reference to internal data model.")
     }
 
-    pub fn fields(&self) -> impl Iterator<Item = Field> + '_ {
+    pub fn fields(&self) -> Vec<Field> {
```

Comment:
Is there a reason why we allocate a vector here?

Reply:
The answer can be "we will fix this later but now it's here because it makes things easier".

Reply:
> we will fix this later but now it's here because it makes things easier

...and prisma-models returns owned vecs a bit everywhere. I also don't like it, but it's a temporary situation.